components/canvas/webgl_paint_task.rs

```diff
@@ -0,0 +1,155 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use azure::azure::AzFloat;
+use azure::azure_hl::{DrawTarget, SurfaceFormat, BackendType, DrawOptions, Filter, DrawSurfaceOptions};
+use canvas_msg::CanvasMsg;
+use compositing::windowing::{WindowMethods};
+use geom::point::Point2D;
+use geom::rect::Rect;
+use geom::size::Size2D;
+
+use gleam::gl;
+use gleam::gl::types::{GLint, GLsizei};
+
+use layers::platform::surface::{NativeGraphicsMetadata, NativePaintingGraphicsContext};
+use util::task::spawn_named;
+
+use skia::SkiaGrGLNativeContextRef;
+use std::borrow::ToOwned;
+use std::sync::mpsc::{channel, Sender};
+use util::vec::byte_swap;
+
+pub struct WebGLPaintTask {
+    size: Size2D<i32>,
+    drawtarget: DrawTarget,
+    native_graphics_context: NativePaintingGraphicsContext
+}
+
+impl WebGLPaintTask {
+    fn new(size: Size2D<i32>, native_graphics_metadata: &NativeGraphicsMetadata) -> WebGLPaintTask {
+        let native_graphics_context = NativePaintingGraphicsContext::from_metadata(&native_graphics_metadata);
+        let draw_target = WebGLPaintTask::create(size, &native_graphics_context);
+        WebGLPaintTask {
+            size: size,
+            drawtarget: draw_target,
+            native_graphics_context: native_graphics_context,
+        }
+    }
+
+    fn create(size: Size2D<i32>,
+              native_graphics_context: &NativePaintingGraphicsContext) -> DrawTarget {
+        let graphics_context = native_graphics_context as *const _ as SkiaGrGLNativeContextRef;
+        let draw_target = DrawTarget::new_with_fbo(BackendType::Skia,
+                                                   graphics_context,
+                                                   size,
+                                                   SurfaceFormat::B8G8R8A8);
```

Comment:
There's no need to create a draw target here, since WebGL has no vector operations. Just store a blob of pixels for now and create the source surface in the paint task. (Eventually we will want to layerize.)